backend/src/test/java/com/zzang/chongdae/offering/service/OfferingServiceTest.java

```diff
@@ -19,59 +25,322 @@ public class OfferingServiceTest extends ServiceTest {
-        OfferingAllResponseItem expected = new OfferingAllResponseItem(offering, offering.toOfferingPrice());
+    @DisplayName("공모 상세 조회")
+    @Nested
+    class GetOfferingDetail {
 
-        // when
-        OfferingAllResponseItem actual = offeringService.getOffering(offering.getId());
+        MemberEntity member;
+        OfferingEntity offering;
 
-        // then
-        assertEquals(expected, actual);
+        @BeforeEach
+        void setUp() {
+            member = memberFixture.createMember("dora");
+            offering = offeringFixture.createOffering(member);
+        }
+
+        @DisplayName("공모 id를 통해 공모 상세를 조회할 수 있다")
+        @Test
+        void should_getOfferingDetail_when_givenOfferingId() {
+            // when
+            OfferingDetailResponse response = offeringService.getOfferingDetail(offering.getId(), member);
+
+            // then
+            assertEquals(offering.getId(), response.id());
+        }
+
+        @DisplayName("존재하지 않는 공모 id를 통해 공모 상세를 조회할 경우 예외가 발생한다.")
+        @Test
+        void should_throwException_when_givenInvalidOfferingId() {
+            // when & then
+            long invalidOfferingId = offering.getId() + 9999;
+
+            assertThatThrownBy(() -> offeringService.getOfferingDetail(invalidOfferingId, member))
+                    .isInstanceOf(MarketException.class);
+        }
+
+        @DisplayName("삭제한 공모 id를 통해 공모 상세를 조회할 경우 예외가 발생한다.")
+        @Test
+        void should_throwException_when_givenDeletedOfferingId() {
+            // given
+            offeringFixture.deleteOffering(offering);
+
+            // when & then
+            assertThatThrownBy(() -> offeringService.getOfferingDetail(offering.getId(), member))
+                    .isInstanceOf(MarketException.class);
+        }
+    }
+
+    @DisplayName("공모 단건 조회")
+    @Nested
+    class GetOffering {
+
+        OfferingEntity offering;
+
+        @BeforeEach
+        void setUp() {
+            MemberEntity member = memberFixture.createMember("ever");
+            offering = offeringFixture.createOffering(member);
+        }
+
+        @DisplayName("공모 id를 통해 공모를 단건 조회할 수 있다")
+        @Test
+        void should_getOffering_when_givenOfferingId() {
+            // given
+            OfferingAllResponseItem expected = new OfferingAllResponseItem(offering, offering.toOfferingPrice());
+
+            // when
+            OfferingAllResponseItem actual = offeringService.getOffering(offering.getId());
+
+            // then
+            assertEquals(expected, actual);
+        }
+
+        @DisplayName("존재하지 않는 공모 id를 통해 공모를 단건 조회할 경우 예외가 발생한다.")
+        @Test
+        void should_throwException_when_givenInvalidOfferingId() {
+            // when & then
+            long invalidOfferingId = offering.getId() + 9999;
+
+            assertThatThrownBy(() -> offeringService.getOffering(invalidOfferingId))
+                    .isInstanceOf(MarketException.class);
```

Comment:
```suggestion
            // given
            long invalidOfferingId = offering.getId() + 9999;

            // when & then
            assertThatThrownBy(() -> offeringService.getOffering(invalidOfferingId))
                    .isInstanceOf(MarketException.class);
```

given은 테스트하기 위한 데이터를 제시하는 것이고
when은 테스트를 수행할 메서드를 수행하는 것이고
then은 수행한 데이터가 기대한 값을 검증하는 것이라고 생각하는데 제가 틀리게 기억하고 있는건지 잘 모르겠어요 😭